DynamicStoragePiles/RestrictContainers.cs

```diff
@@ -0,0 +1,298 @@
+﻿using System.Collections.Generic;
+using HarmonyLib;
+
+namespace DynamicStoragePiles {
+
+    /// <summary>
+    ///     Class to patch Valheim inventory methods so that the specified
+    ///     containers can only contain items of a single type.
+    /// </summary>
+    [HarmonyPatch(typeof(Inventory))]
+    internal static class RestrictContainers {
+        private static string _loadingContainer;
+        private static string _targetContainer;
+        private static string _allowedItem;
+        private static Dictionary<string, string> _allowedItemsByContainer = new Dictionary<string, string>();
```

Comment:
The second Dictionary seems a bit convoluted, the Plugin allowedItemsByContainer could be made public and accessed from the class.

Reply:
Yeah, it is a bit convoluted, I had used that approach since I wrote this code in parallel with what I wrote for OdinsFoodBarrels so I tried to make the functionality of RestrictContainers as encapsulated and portable as possible. For this use case making the dictionary from Plugin internal or public is a better solution.